packages/components/src/date-time/time.js

```diff
@@ -120,46 +180,129 @@ class TimePicker extends Component {
-				/>
-				<span className="components-time-picker__separator">:</span>
-				<input
-					className="components-time-picker__input"
-					type="text"
-					value={ minutes }
-					onChange={ this.onChangeMinutes }
-					onBlur={ this.updateMinutes }
-				/>
-				{ is12Hour && <div>
-					<Button
-						isDefault
-						className="components-time-picker__am-button"
-						isToggled={ am === 'AM' }
-						onClick={ this.updateAmPm( 'AM' ) }
-					>
-						{ __( 'AM' ) }
-					</Button>
-					<Button
-						isDefault
-						className="components-time-picker__pm-button"
-						isToggled={ am === 'PM' }
-						onClick={ this.updateAmPm( 'PM' ) }
-					>
-						{ __( 'PM' ) }
-					</Button>
-				</div> }
-			</div>
+			<fieldset
+				className={
+					is12Hour ? 'components-datetime__time components-datetime__time--12-hour' : 'components-datetime__time components-datetime__time--24-hour'
+				}
+			>
+				<legend>{ __( 'Publish at' ) }</legend>
+				<fieldset className="invisible">
+					<legend className="invisible">{ __( 'Date' ) }</legend>
+					<div className="components-datetime__time__form">
+						<div className="components-datetime__time__form__container components-datetime__time__form__container--month">
+							<select
+								aria-label={ __( 'Month' ) }
+								name="month"
+								value={ month }
+								onChange={ this.onChangeMonth }
+								onBlur={ this.updateMonth }
+							>
+								<option value="01">{ __( 'January' ) }</option>
+								<option value="02">{ __( 'February' ) }</option>
+								<option value="03">{ __( 'March' ) }</option>
+								<option value="04">{ __( 'April' ) }</option>
+								<option value="05">{ __( 'May' ) }</option>
+								<option value="06">{ __( 'June' ) }</option>
+								<option value="07">{ __( 'July' ) }</option>
+								<option value="08">{ __( 'August' ) }</option>
+								<option value="09">{ __( 'September' ) }</option>
+								<option value="10">{ __( 'October' ) }</option>
+								<option value="11">{ __( 'November' ) }</option>
+								<option value="12">{ __( 'December' ) }</option>
+							</select>
+						</div>
+						<div key="day-field" className="components-datetime__time__form__container components-datetime__time__form__container--day">
+							<input
+								aria-label={ __( 'Day' ) }
+								type="number"
+								name="day"
+								value={ day }
+								step={ 1 }
+								min={ 1 }
+								onChange={ this.onChangeDay }
+								onBlur={ this.updateDay }
+							/>
+						</div>
+						<div key="year-field" className="components-datetime__time__form__container components-datetime__time__form__container--year">
+							<input
+								aria-label={ __( 'Year' ) }
+								type="number"
+								name="year"
+								step={ 1 }
+								value={ year }
+								onChange={ this.onChangeYear }
+								onBlur={ this.updateYear }
+							/>
+						</div>
+					</div>
+				</fieldset>
+
+				<fieldset>
+					<legend className="invisible">{ __( 'Time' ) }</legend>
```

Comment:
Where does `.invisible` come from? It doesn't adhere to the Gutenberg class naming convention. Is it something being inherited from wp-admin? In which case, as everything in `@wordpress/components` is meant to be usable in a standalone context, we should not assume to be able to inherit this style. If we need it, we should recreate it within the package-scoped styles.

Reply:
It's in `date-time/style.scss`. I'd agree about the naming. Related: worth noting there are a few cases in Gutenberg where the `screen-reader-text` CSS class is used and, as far as I know, it's inherited from core.